### PR TITLE
implement `dpnp.logsumexp` and `dpnp.reduce_hypot`

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, `dpnp.logsumexp` and `dpnp.reduce_hypot` are implemented using their `dpctl` counterpart.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_trigonometric.py [hunk 1]
```diff
@@ -1331,6 +1340,83 @@ def logaddexp(
...
+    Parameters
+    ----------
+    x : {dpnp_array, usm_ndarray}
+        Input array.
+    axis : int or tuple of ints, optional
+        Axis or axes along which values must be computed. If a tuple
+        of unique integers, values are computed over multiple axes.
+        If ``None``, the result is computed over the entire array.
+        Default: ``None``.
+    out :  {dpnp_array, usm_ndarray}, optional
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1356]:
```suggestion
    out : {dpnp_array, usm_ndarray}, optional
```

File: dpnp/dpnp_iface_trigonometric.py [hunk 2]
```diff
@@ -1331,6 +1340,83 @@ def logaddexp(
...
+    -------
+    out : dpnp.ndarray
+        An array containing the results. If the result was computed over
+        the entire array, a zero-dimensional array is returned. The returned
+        array has the data type as described in the `dtype` parameter
+        description above.
+
+    Limitations
+    -----------
+    Input array is only supported as either scalar, :class:`dpnp.ndarray`
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1391]: Based on the code, scalar is not supported

File: dpnp/dpnp_iface_trigonometric.py [hunk 3]
```diff
@@ -1331,6 +1340,83 @@ def logaddexp(
...
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        An array containing the results. If the result was computed over
+        the entire array, a zero-dimensional array is returned. The returned
+        array has the data type as described in the `dtype` parameter
+        description above.
+
+    Limitations
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1389]: in general, we can remove the limitation block here:

- supported type of input array is already stated at line#1349
- support of real-valued data types only can be stated at the description at line#1350

File: dpnp/dpnp_iface_trigonometric.py [hunk 4]
```diff
@@ -1363,6 +1449,81 @@ def reciprocal(x1, **kwargs):
...
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1498]: the same comment here

File: tests/helper.py [hunk 5]
```diff
@@ -33,7 +33,7 @@ def assert_dtype_allclose(
 
     list_64bit_types = [numpy.float64, numpy.complex128]
     is_inexact = lambda x: dpnp.issubdtype(x.dtype, dpnp.inexact)
-    if is_inexact(dpnp_arr) or is_inexact(numpy_arr):
+    if is_inexact(dpnp_arr) and is_inexact(numpy_arr):
```
Comment (antonwolfy) [inline tests/helper.py:36]: What is the reason for change? I assumed we can use `assert_array_equal` only when both input are not of inexact dtype.

File: dpnp/dpnp_iface_trigonometric.py [hunk 6]
```diff
@@ -1331,6 +1340,83 @@ def logaddexp(
     )
 
 
+def logsumexp(x, axis=None, out=None, dtype=None, keepdims=False):
+    """
+    Calculates the logarithm of the sum of exponentials of elements in the input array.
+
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1346]: do we need to note that it is equivalent of `numpy.logaddexp.reduce`?

File: dpnp/dpnp_iface_trigonometric.py [hunk 7]
```diff
@@ -1363,6 +1449,81 @@ def reciprocal(x1, **kwargs):
     return call_origin(numpy.reciprocal, x1, **kwargs)
 
 
+def reduce_hypot(x, axis=None, out=None, dtype=None, keepdims=False):
+    """
+    Calculates the square root of the sum of squares of elements in the input array.
+
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:1455]: the same question about `numpy.hypot.reduce` here

File: dpnp/dpnp_iface_trigonometric.py [hunk 8]
```diff
@@ -98,9 +100,11 @@
     "log1p",
     "log2",
     "logaddexp",
+    "logsumexp",
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_trigonometric.py:103]: need to update .rst file to generate documentation for new functions

File: tests/helper.py (hunk 5 above)
Comment (vtavana) [reply tests/helper.py:36]: In calculating `tol`, we use `finfo` for both of them which is only defined for `inexact` data types.

**Review by antonwolfy — APPROVED** [review]
Thank you @vtavana ! No more comments from me.

**Pull request merged**